src/v/datalake/proto_to_arrow_interface.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include <arrow/api.h>
+#include <arrow/array/array_base.h>
+#include <arrow/type_fwd.h>
+
+#include <memory>
+#include <stdexcept>
+
+namespace google::protobuf {
+class Message;
+}
+
+namespace datalake::detail {
+
+// This interface is used to convert a set of Protobuf messages to an Arrow
+// Array.
+class proto_to_arrow_interface {
+public:
+    proto_to_arrow_interface(const proto_to_arrow_interface&) = delete;
+    proto_to_arrow_interface(proto_to_arrow_interface&&) = delete;
+    proto_to_arrow_interface& operator=(const proto_to_arrow_interface&)
+      = delete;
+    proto_to_arrow_interface& operator=(proto_to_arrow_interface&&) = delete;
+
+    proto_to_arrow_interface() = default;
+    virtual ~proto_to_arrow_interface() = default;
+
+    /// Return an Arrow field descriptor for this Array. Used for building
+    /// A schema.
```

Comment:
nit: it doesn't look like this is used anymore (no longer needed since struct converters can return their arrow schemas?)